src/Framework/test/PackageTests.cs

```diff
@@ -0,0 +1,135 @@
+            var outputPath = Path.Combine(_packageLayoutRoot, Path.GetFileNameWithoutExtension(package));
+            ZipFile.ExtractToDirectory(package, outputPath);
+        }
+    }
+
+    [Fact]
+    public void PackageAssembliesHaveExpectedAssemblyVersions()
+    {
+        // Do nothing if this is a dev build
+        if (!TestData.VerifyPackageAssemblyVersions() && !SkipOnHelixAttribute.OnHelix())
+        {
+            return;
+        }
+
+        var versionStringWithoutPrereleaseTag = TestData.GetSharedFxVersion().Split('-', 2)[0];
+        var expectedVersion = Version.Parse(versionStringWithoutPrereleaseTag);
+
+        foreach (var packageDir in Directory.GetDirectories(_packageLayoutRoot))
+        {
+            // Don't test the Shared Framework or Ref pack
+            if (packageDir.StartsWith("Microsoft.AspNetCore.App", StringComparison.OrdinalIgnoreCase))
+            {
+                continue;
+            }
+            // Test lib assemblies
+            var packageAssembliesDir = Path.Combine(packageDir, "lib");
+            if (Directory.Exists(packageAssembliesDir))
+            {
+                foreach (var tfmDir in Directory.GetDirectories(packageAssembliesDir))
+                {
+                    var tfm = new DirectoryInfo(tfmDir).Name;
+                    foreach (var assembly in Directory.GetFiles(tfmDir, "*.dll"))
+                    {
+                        using var fileStream = File.OpenRead(assembly);
+                        using var peReader = new PEReader(fileStream, PEStreamOptions.Default);
+                        var reader = peReader.GetMetadataReader(MetadataReaderOptions.Default);
+                        var assemblyVersion = reader.GetAssemblyDefinition().Version;
+
+                        // net & netstandard assembly versions should all match Major.Minor.0.0
+                        // netfx assembly versions should match Major.Minor.Patch.0
+                        Assert.Equal(expectedVersion.Major, assemblyVersion.Major);
+                        Assert.Equal(expectedVersion.Minor, assemblyVersion.Minor);
+                        if (IsNetFx(tfm))
+                        {
+                            Assert.Equal(expectedVersion.Build, assemblyVersion.Build);
+                        }
+                        else
+                        {
+                            Assert.Equal(0, assemblyVersion.Build);
+                        }
+                        Assert.Equal(0, assemblyVersion.Revision);
+                    }
+                }
+            }
+
+            // Test tool assemblies
+            var packageToolsDir = Path.Combine(packageDir, "tools");
+            if (Directory.Exists(packageToolsDir))
+            {
+                var assemblies = Directory.GetFiles(packageToolsDir, "*.dll", SearchOption.AllDirectories)
+                    .Where(f => !Path.GetFileNameWithoutExtension(f).Contains("System.", StringComparison.OrdinalIgnoreCase) &&
```

Comment:
This is ugly, @dougbu do you know if there's a cleaner way w/ LINQ to return everything from GetFiles where FileNameWithoutExtension isn't in the below list of things?

Reply:
One idea would be to `Split(...)` the name using `.` and `_` then check whether the first segment is in `{ "aspnetcorev2", "Azure", "Microsoft", "Newtonsoft", "System" }`. I'm sure @davidfowl or @halter73 would have a non-allocating suggestion but this is test code 😀